src/main/java/dev/j3fftw/litexpansion/items/Wrench.java

```diff
@@ -0,0 +1,144 @@
+
+    private static final int[] cargoSlots = { 19, 20, 21, 28, 29, 30, 37, 38, 39 };
+    private static final String[] specialBlockIds = { SlimefunItems.CARGO_INPUT_NODE.getItemId(), SlimefunItems.CARGO_OUTPUT_NODE.getItemId(), SlimefunItems.CARGO_OUTPUT_NODE_2.getItemId(), SlimefunItems.CARGO_CONNECTOR_NODE.getItemId(), SlimefunItems.SMALL_CAPACITOR.getItemId(), SlimefunItems.MEDIUM_CAPACITOR.getItemId(), SlimefunItems.LARGE_CAPACITOR.getItemId(), SlimefunItems.BIG_CAPACITOR.getItemId(), SlimefunItems.CARBONADO_EDGED_CAPACITOR.getItemId(), SlimefunItems.TRASH_CAN.getItemId() };
+
+    public Wrench() {
+        super(Items.LITEXPANSION, Items.WRENCH, RecipeType.ENHANCED_CRAFTING_TABLE, new ItemStack[] {
+                Items.REFINED_IRON, null, Items.REFINED_IRON,
+                null, Items.REFINED_IRON, null,
+                null, Items.REFINED_IRON, null
+        });
+
+        Bukkit.getPluginManager().registerEvents(this, LiteXpansion.getInstance());
+    }
+
+    @Override
+    public ItemUseHandler getItemHandler() {
+        return e -> e.setUseBlock(Event.Result.DENY);
+    }
+
+    @EventHandler
+    public void onWrenchInteract(PlayerInteractEvent e) {
+        Player p = e.getPlayer();
+        // Check if player has wrench and is left clicking block
+        // Can't use offhand because a player can offhand the wrench to escape the event
+        if (isItem(e.getItem()) && !isItem(p.getInventory().getItemInOffHand())
+            && e.getAction().toString().endsWith("_BLOCK")
+        ) {
+            e.setCancelled(true);
+            Block block = e.getClickedBlock();
+            SlimefunItem slimefunBlock = BlockStorage.check(block);
+
+            // Check if slimefunBlock is not a machine or a cargo component
+            if (slimefunBlock == null
+                || (!(slimefunBlock instanceof EnergyNetComponent)
+                && !slimefunBlock.getID().startsWith("CARGO_NODE")
+                && !slimefunBlock.getID().equals(SlimefunItems.TRASH_CAN.getItemId()))
+            ) {
+                p.sendMessage(ChatColor.RED + "Hey buddy boy, this can only be electric machines, generators, capacitors, and cargo nodes!");
+                return;
+            }
+
+            // Check if the config states that player needs wrench
+            if (Constants.MACHINE_BREAK_REQUIRES_WRENCH) {
+                double failChance = Math.random();
+                if (failChance <= Constants.WRENCH_FAIL_CHANCE) {
+                    dropBlock(e, p, block, slimefunBlock, true);
+                } else {
+                    dropBlock(e, p, block, slimefunBlock, false);
+                }
+            } else {
+                dropBlock(e, p, block, slimefunBlock, false);
+            }
+            damageItem(p, e.getItem());
+        }
+    }
+
+    public static void dropBlock(Event e, Player p, Block block, SlimefunItem slimefunBlock, boolean failed){
+        ItemStack slimefunBlockDrop = slimefunBlock.getItem();
```

Comment:
Also... NEVER NEVER NEVER NEVER NEVER use the original Item, always do `.clone()`

Reply:
I specifically added a `WrongItemStackException` whenever things like that lead to issues.